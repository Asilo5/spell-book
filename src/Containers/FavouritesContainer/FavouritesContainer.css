.favourites-container {
    background-image: url('https://vignette.wikia.nocookie.net/pottermore/images/b/b6/Divination.png/revision/latest?cb=20121221000858');
    height: 100vh;
    background-size: cover;
}

.favourites {
  text-align: center;
}