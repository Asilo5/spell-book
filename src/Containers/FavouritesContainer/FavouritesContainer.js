import React from 'react';
import { connect } from 'react-redux';
import SpellBook from '../SpellBook/SpellBook';
import { Link } from 'react-router-dom';
import './FavouritesContainer.css';

const FavouritesContainer = ({favouriteSpells, spellBook}) => {
   const findSpell = favouriteSpells.map((id) => {
    return spellBook.find((spell) => spell._id === id);
  });

  const favSpell = findSpell.map((spell) => {
    return <SpellBook {...spell} />
  })

    return (
      <section className='favourites-container'>
          <Link to='/spells'>
            <button className='back-btn-fav' >Back</button>
          </Link>
          <h2 className='favourites'>Favourite Spells</h2>
          <div className='fav-spells'>
            {favSpell}
          </div>
      </section>
    )
}

export const mapStateToProps = ({favouriteSpells, spellBook}) => ({
    favouriteSpells,
    spellBook
})

export default connect(mapStateToProps)(FavouritesContainer);