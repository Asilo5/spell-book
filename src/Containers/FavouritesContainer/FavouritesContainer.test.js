import React from 'react';
import { FavouritesContainer, mapStateToProps } from './FavouritesContainer';
import { shallow } from 'enzyme';

describe('FavouritesContainer', () => {

    describe('favourites', () => {

       let mockFavouriteSpell = [
        { _id: '5b74f3653228320021ab628b',
        spell: 'Lumos',
        type: 'Spell',
        effect: 'creates light at wand tip',
        __v: 0 }
       ];

       let mockSpellBook = [
        { _id: '5b74f3583228320021ab628a',
        spell: 'Locomotor Mortis',
        type: 'Curse',
        effect: 'locks opponent\'s legs',
        __v: 0 },
      { _id: '5b74f3653228320021ab628b',
        spell: 'Lumos',
        type: 'Spell',
        effect: 'creates light at wand tip',
        __v: 0 } 
       ];

       let wrapper = shallow( <FavouritesContainer 
               favouriteSpells={mockFavouriteSpell}
                spellBook={mockSpellBook}
       /> )
       
      it('should match snapshot with all information passing in correctly', () => {
         expect(wrapper).toMatchSnapshot();
      })
    })

    describe('mapStateToProps', () => {
        it('should return a spellBook and favouriteSpell object', () => {
            const mockState = {
               "favouriteSpells": [   
                { _id: '5b74f3653228320021ab628b',
                   spell: 'Lumos',
                   type: 'Spell',
                   effect: 'creates light at wand tip',
                   __v: 0 }
                ],
               "spellBook": [
                  { _id: '5b74f3653228320021ab628b',
                  spell: 'Lumos',
                  type: 'Spell',
                  effect: 'creates light at wand tip',
                __v: 0 }
               ],
            }
            const expected = {
               "favouriteSpells": [
                { _id: '5b74f3653228320021ab628b',
                spell: 'Lumos',
                type: 'Spell',
                effect: 'creates light at wand tip',
                __v: 0 }
               ],
                "spellBook": [
                    { _id: '5b74f3653228320021ab628b',
                   spell: 'Lumos',
                   type: 'Spell',
                   effect: 'creates light at wand tip',
                   __v: 0 }
                ],
            }
            const mappedProps = mapStateToProps(mockState);
            expect(mappedProps).toEqual(expected);
          });
    })

})