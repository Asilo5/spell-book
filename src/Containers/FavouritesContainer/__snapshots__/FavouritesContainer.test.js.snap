// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavouritesContainer favourites should match snapshot with all information passing in correctly 1`] = `
<section
  className="favourites-container"
>
  <Link
    to="/spells"
  >
    <button
      className="back-btn-fav"
    >
      Back
    </button>
  </Link>
  <h2
    className="favourites"
  >
    Favourite Spells
  </h2>
  <div
    className="fav-spells"
  >
    <Connect(SpellBook) />
  </div>
</section>
`;
